src/orquestra/sdk/dremio/_api.py

```diff
@@ -0,0 +1,55 @@
+################################################################################
+# © Copyright 2023 Zapata Computing Inc.
+################################################################################
+from dataclasses import dataclass
+
+import certifi
+
+from ._flight_facade import FlightCallOptions, FlightClient, FlightDescriptor
+
+
+class DremioClient:
+    @dataclass(frozen=True)
+    class Config:
```

Comment:
@alexjuda out-of-curiosity: With this dataclass implementation, we can create an instance by calling `DremioClient.Config(**kwargs)`, I was wondering what reasons are behind this choice given that we can also create an instance by simply running `DremioClient(**kwargs)` if we omit nested Config class?

Reply:
1. [Dependency injection](https://en.wikipedia.org/wiki/Dependency_injection). `DremioClient`'s init accepts `flight_client` object as an argument. Good for code modularity, bad for user convenience. Instead, there's a convenience [factory method](https://en.wikipedia.org/wiki/Factory_method_pattern) for human users.
2. There's a convenience `DremioClient.from_config` that's good for human users. We could make it accept `user`, `password`, `host`, and `port` arguments. However:
3. The original code included a config class, closely related to `DremioClient` (can't link it because this is a public repo). As far as I understood this feature request's requirements, we should be able to easily serialize/deserialize DremioClient configuration. Hence, the dataclass.